enhancements/network/bgp-overview.md

```diff
@@ -0,0 +1,173 @@
+Service networks via BGP.
+
+If we wanted to offer this, it would not be very difficult.  At least with
+OpenShift-SDN and OVN-Kubernetes, Nodes are already set up to handle traffic
+destined for Pod or Service cluster IP addresses if traffic were to arrive
+destined for those addresses.
+
+To enable this using BGP, a BGP speaker must run on each Node where we would
+like to serve as a router into the cluster network.  Each Node can publish a
+route for the Pod and Service network and upstream routers may use ECMP to
+distribute traffic among the Nodes.
+
+A future enhancement needs to be filed to pursue this integration in more
+detail.
+
+### Virtual Network Interconnect
+
+The Pod network is by default an isolated virtual network.  There is sometimes
+a desire to interconnect multiple virtual networks together.  This could be
+between multiple Kubernetes clusters, but it doesn't have to be the case.
+
+BGP EVPN (Ethernet VPN, or Virtual Private Network) is a BGP technology that is
+discussed in the context of this use case.  BGP EVPN can be used to advertise
+MAC addresses and IP-MAC address bindings.  BGP EVPN is not a full solution to
+this use case by itself, as it is a technology that can help implement part of
+the control plane.  There are more details that must be explored, including
+what the data plane looks like between the two clusters.
+
+This is a significantly non-trivial effort.  Future enhancements are needed to
+discuss this in much more detail.
+
+## Common Design Considerations
+
+This section summarizes the common design considerations for OpenShift in
+support of BGP related features.
+
+### Avoid L2 (Layer 2) Domain Assumptions
+
+We must avoid the assumption of Nodes residing on the same L2 domain as much as
+possible.  Every OpenShift Node should only assume L3 connectivity to every
+other Node.
+
+One example that should be avoided is any use of keepalived, as all Nodes
+running keepalived to manage a VIP (Virtual IP address) must reside on the same
+L2 segment since ARP (IPv4) or NDP (IPv6) is used to announce the new location
+of the VIP when it moves to a new Node.
+
+### Avoid any Hard Requirements on BGP
+
+Any use of BGP in OpenShift should remain optional.  We have some customers
+that do not want to use BGP at all, either for some technical or policy reason.
+We may explore using it to add optional features, but it should not be required
+for any OpenShift clusters.
+
+For example, we are considering [adding MetalLB][3] to OpenShift to support
+Services of `type=LoadBalancer`.  While MetalLB can use BGP, it also has an
+alternative `layer2` mode that can provide the same functionality, though with
+a different set of limitations.
+
+## BGP Resources
+
+* https://blog.cdemi.io/beginners-guide-to-understanding-bgp/
+* https://www.ciscopress.com/articles/article.asp?p=2756480
+* https://en.wikipedia.org/wiki/Border_Gateway_Protocol
+
+[1]: https://projectcalico.org
```

Comment:
This should be dropped, as Calico is no longer mentioned in the proposal.

Reply:
good catch, thanks!